src/Api/Customer.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Billogram\Api;
+
+use Billogram\Api;
+use Billogram\Exception\Domain\ValidationException;
+use Billogram\Exception\InvalidArgumentException;
+use Billogram\Model\Customer\Customer as Model;
+use Billogram\Model\Customer\Customers;
+use Billogram\Api\HttpApi;
+
+class Customer extends HttpApi
+{
+    /**
+     * @param array $param
+     * @return string|array
+     * @link
+     */
+    public function search(array $param = ['page'=>1, 'page_size'=> 100 ])
+    {
+        $response= $this->httpget('/customer', $param);
+
+        if (!$this->hydrator) {
+            return $response;
+        }
+
+        // Use any valid status code here
+        if ($response->getStatusCode() !== 200) {
+            $this->handleErrors($response);
+        }
+        return $this->hydrator->hydrate($response, Customers::class);
+
+    }
+
+    /**
+     *
+     * @param int   $customerNo
+     * @param array $param
+     *
+     *
+     * @link https://billogram.com/api/documentation#customers_fetch
+     * @return \Billogram\Model\Customer\Customer
+     */
+    public function fetch(int $customerNo, array $param = [])
+    {
+        if (empty($customerNo)) {
+            throw new InvalidArgumentException('Id cannot be empty');
+        }
+        $response = $this->httpGet('/customer/'.$customerNo, $param);
+
+        if (!$this->hydrator) {
+            return $response;
+        }
+        // Use any valid status code here
+        if ($response->getStatusCode() !== 200) {
+            $this->handleErrors($response);
+        }
+        return $this->hydrator->hydrate($response, Model::class);
+
+    }
+
+    /**
+     * @param Model $costumer
+     *  @link https://billogram.com/api/documentation#customers_create
+     */
+    public function create(Model $costumer)
+    {
+        if (empty($costumer)) {
```

Comment:
We call toArray or (arrray)$custumer to cast $customer till an array then we test if it is empty else i can delete it